### PR TITLE
Menu 1

### PR DESCRIPTION
**Pull request by ryliemn**
basic menu appearance, filling in selects from data retreived from endpoint

### CODE REVIEW

File: jQuery/honeyHeadquarters.js [hunk 1]
```diff
@@ -1,4 +1,27 @@
 // $.ajax('https://sheetsu.com/apis/v1.0/64b5c3f8').done(function(data) {
-$.ajax('http://localhost:3000/citizens').done(function(data) {
-  console.log(data);
+$.ajax('http://localhost:3000/citizens').done(function(citizens) {
+  function getUniqueValuesForField(field, citizens) {
+    const values = ['All'];
```
Comment (stffrd) [inline jQuery/honeyHeadquarters.js]: Ooh. `const` is a no-no. You're making assumptions about your environment here. Chrome supports `const`, and `let` and all your other fancy es6 stuff. 

best to use `var` for maximum compatibility between browsers. 
Otherwise you risk just straight up erroring in other browsers.

File: jQuery/honeyHeadquarters.js [hunk 2]
```diff
@@ -1,4 +1,27 @@
...
+    citizens.forEach((c) => {
+      let v = c[field];
```
Comment (stffrd) [inline jQuery/honeyHeadquarters.js]: Same thing here. `var` is better for today's world.

File: jQuery/honeyHeadquarters.js [hunk 3]
```diff
@@ -1,4 +1,27 @@
...
+      if (!values.includes(v)) {
```
Comment (stffrd) [inline jQuery/honeyHeadquarters.js]: `Array.includes` has no support in [internet explorer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes)

If you need a browser friendly way to check, you can use `if(values.indexOf(v) > -1)` or `if(!~values.indexOf(v))`, the `~` operator is a bit shifter. makes a -1 a 0, makes any other number a non-zero number, which evaluates to true.

File: jQuery/honeyHeadquarters.js [hunk 4]
```diff
@@ -1,4 +1,27 @@
 // $.ajax('https://sheetsu.com/apis/v1.0/64b5c3f8').done(function(data) {
-$.ajax('http://localhost:3000/citizens').done(function(data) {
-  console.log(data);
+$.ajax('http://localhost:3000/citizens').done(function(citizens) {
+  function getUniqueValuesForField(field, citizens) {
+    const values = ['All'];
+    citizens.forEach((c) => {
```
Comment (stffrd) [inline jQuery/honeyHeadquarters.js]: You're using a fat arrow syntax here. This is es6. ya can't do it if you want this to work in more than just chrome/firefox. `function() { ... }` won't kill you 💯

File: jQuery/honeyHeadquarters.js [hunk 5]
```diff
@@ -1,4 +1,27 @@
 // $.ajax('https://sheetsu.com/apis/v1.0/64b5c3f8').done(function(data) {
-$.ajax('http://localhost:3000/citizens').done(function(data) {
-  console.log(data);
+$.ajax('http://localhost:3000/citizens').done(function(citizens) {
```
Comment (stffrd) [inline jQuery/honeyHeadquarters.js]: You use fat arrows down there and `function() { ... }` up here as a lambda? Use `function() { ... }` almost exclusively unless you have es6 tooling.

File: jQuery/honeyHeadquarters.js [hunk 6]
```diff
@@ -1,4 +1,27 @@
...
+    return values;
+  }
+
+  function populateSelectWithOptions(select, options) {
+    options.forEach((option) => {
+      $(select).append($("<option>", { value: option, html: option }));
+    });
+  }
+
+  const characters = getUniqueValuesForField('character', citizens);
```
Comment (stffrd) [inline jQuery/honeyHeadquarters.js]: NO.

File: jQuery/honeyHeadquarters.js [hunk 7]
```diff
@@ -1,4 +1,27 @@
...
+  const hometowns = getUniqueValuesForField('hometown', citizens);
```
Comment (stffrd) [inline jQuery/honeyHeadquarters.js]: CUT IT OUT.

File: jQuery/honeyHeadquarters.js [hunk 8]
```diff
@@ -1,4 +1,27 @@
...
+  const sizes = getUniqueValuesForField('size', citizens);
```
Comment (stffrd) [inline jQuery/honeyHeadquarters.js]: VAR, YOU DINGUS.

File: jQuery/index.html [hunk 9]
```diff
@@ -1,10 +1,25 @@
...
   <head>
+    <link rel="stylesheet" type="text/css" href="styles.css">
+
     <title>Honey Headquarters</title>
   </head>
   <body>
-    Hello wolr
+    <div id="hh-menu">
+      <h1>Honey Headquarters</h1>
+
```
Comment (stffrd) [inline jQuery/index.html:12]: an interesting way to structure labels with form elements can actually be implicit rather than explicit. You're explicitly declaring that a `label` is `for` another element. (which is valid) but you can also do this:

```
<label class="menu-label">
  <select id="character-select"></select>
</label>
```

The class on this label isn't necessary, just an idea. but semantically this is still valid, as the form element is **implicitly** targeted by the label by virtue of being contained inside of it.

File: jQuery/styles.css [hunk 10]
```diff
@@ -0,0 +1,30 @@
+body {
```
Comment (stffrd) [inline jQuery/styles.css]: Unless you're using a reset, targeting elements generically is generally stupid and bad.

File: jQuery/styles.css [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+#hh-menu {
+  border-right: solid 1px black;
+  position: fixed;
+  width: 240px;
+  height: 100%;
+}
+
+#hh-menu > select {
```
Comment (stffrd) [inline jQuery/styles.css]: remember up there when I put a class of `menu-label` on a label? you could do the same with `select`

```
.menu-option {
  width: 80%;
  margin-left: 15%;
 margin-right: 5%;
}
```

What you're doing with these margins is... eugh. 
but we'll get to that in another pass :)

File: jQuery/styles.css [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+  width: 80%;
+  margin-left: 15%;
+  margin-right: 5%;
+}
+
+#hh-menu > label {
```
Comment (stffrd) [inline jQuery/styles.css]: Same as above, consider using a class for all your labels. direct child selectors are slower in older browsers (or sometimes unsupported entirely (in ie8))

**Review by stffrd — COMMENTED** [review]
Off to a decent start. I hate camel case so I'll be trying super hard to make you change your function names. Let's take this real slow, though!

**Comment by ryliemn** [comment]
Regarding any ES6 related issues, I'm aware of the incompatibility, but I mainly just figured we wouldn't really care for purposes of this since it's just the prototype before moving on to an actual framework with real tooling and a transpile step. I just want it to be known I AM aware that it's best practice to target ES3/ES5 for browser code

File: jQuery/honeyHeadquarters.js (hunk 5 above)
Comment (ryliemn) [reply jQuery/honeyHeadquarters.js]: Would it damage your soul if, instead of changing everything below to `function() {...}`, I changed this to arrow syntax?

File: jQuery/honeyHeadquarters.js (hunk 1 above)
Comment (ryliemn) [reply jQuery/honeyHeadquarters.js]: see main thread for response to es6 related things

File: jQuery/styles.css (hunk 11 above)
Comment (ryliemn) [reply jQuery/styles.css]: The margin stuff felt wrong and I specifically hoped you would address it. I'm interested to learn the better way of doing it

File: jQuery/honeyHeadquarters.js (hunk 3 above)
Comment (ryliemn) [reply jQuery/honeyHeadquarters.js]: Fuck IE

Comment (ryliemn) [reply jQuery/honeyHeadquarters.js]: I just want you to know I'm going to be as belligerent as possible throughout this whole process

Comment (ryliemn) [reply jQuery/honeyHeadquarters.js]: LOVE YOU

**Comment by stffrd** [comment]
Ah. My contention is that we make a prototype that would theoretically work _in all browsers_. 

I feel like the "shortcut" is negligibly beneficial for a larger drawback. in this case.

File: jQuery/styles.css (hunk 11 above)
Comment (stffrd) [reply jQuery/styles.css]: The better way involves float based layout, ends up being much cleaner.

File: jQuery/honeyHeadquarters.js (hunk 3 above)
Comment (stffrd) [reply jQuery/honeyHeadquarters.js]: Dude. do it. I'm going to hammer the bad habits right out of you.

File: jQuery/honeyHeadquarters.js [hunk 13]
```diff
@@ -1,25 +1,25 @@
 // $.ajax('https://sheetsu.com/apis/v1.0/64b5c3f8').done(function(data) {
 $.ajax('http://localhost:3000/citizens').done(function(citizens) {
```
Comment (stffrd) [inline jQuery/honeyHeadquarters.js]: So as far as everything goes in this `$.ajax` function, you're doing a lot. In fact, you're defining all of your functions inside of the callback function. 

Perfectly valid, but what happens when the callback executes? what if you want to make a call to another endpoint of some other api and you want to reuse these functions?

yeah that's right. you can't! D:

let's repurpose this file a little bit and think about structure:

let's call this file something that denotes that it's the prime mover, e.g. `index.js` or `main.js` or `driver.js`
Secondarily, let's move your ajax block into another file called `ajax.js`. This gives a cleaner separation of concerns, as we know we need to do two things in the driver file:

1: use ajax to get our stuff
2: use jQuery to populate our DOM.

File: jQuery/index.html [hunk 14]
```diff
@@ -9,12 +10,20 @@
     <div id="hh-menu">
       <h1>Honey Headquarters</h1>
 
-      <label for="character-select">Character</label>
-      <select id="character-select"></select>
-      <label for="hometown-select">Hometown</label>
-      <select id="hometown-select"></select>
-      <label for="size-select">Size</label>
-      <select id="size-select"></select>
+      <label class="menu-label">
```
Comment (stffrd) [inline jQuery/index.html:13]: This is much better. You may choose to add classes or give more descriptors if you think they'll help :)

Comment (stffrd) [reply jQuery/index.html:13]: e.g. `label class="menu-label character-label"` etc. etc.

File: jQuery/index.html [hunk 15]
```diff
@@ -9,12 +10,20 @@
     <div id="hh-menu">
```
Comment (stffrd) [inline jQuery/index.html]: I'm not a fan of too many id's on a page, for what it's worth. I get that you'll have a single menu, and so an ID is fine, but you really don't have to qualify it with a namespace (`hh-`), we know that this page is the homepage for honey headquarters.

File: jQuery/styles.css [hunk 16]
```diff
@@ -1,27 +1,19 @@
...
+  padding: 20px;
 }
 
-#hh-menu > label {
-  margin-left: 5%;
+.menu-select {
+  width: 90%;
 }
 
 #hh-content {
```
Comment (stffrd) [inline jQuery/styles.css]: Hard fucking stop. 

Jesus christ. 

one of the first things you learn is how to center a site. Apparently you didn't get that (shame on whoever didn't teach you) So now we're gonna teach you. 

So, you know how you have this structure?

```
<div id="hh-menu">
...
</div>
<div id="hh-content">
...
</div>
```

Wrap the two of those in a div with a single id of `content-wrapper`

remove `hh-menu`s ID, as well as `hh-content`, and replace them with classes `class="menu"` and `class="content"` respectively. 

Now, instead of 

```
#hh-content {
...
}
```

You want this:

#content-wrapper {
  width: 60%; /\* or 80% or whatever */
  margin: 0 auto; 
}

Width shortens the divs length to not take up the whole window, and `margin: 0 auto` will actually center the element in the window. the 'auto' property denotes that its left and right margin calculation should be left up to the browser. In these cases, the browser gives equal margin to both sides. Effectively centering it.  [Here's a CSS tricks about it](https://css-tricks.com/snippets/css/centering-a-website/)

**Review by stffrd — COMMENTED** [review]
Getting there!

File: jQuery/honeyHeadquarters.js (hunk 13 above)
Comment (ryliemn) [reply jQuery/honeyHeadquarters.js]: not really sure I understand what you mean here. Move the ENTIRE block into another file, or just the function declarations? the latter makes more sense to me, since then I can reuse the functions elsewhere which addresses the problem you brought up, and I know that the last six lines of the ajax block are only relevant there, but it seems like you're suggesting an alternative

File: jQuery/styles.css (hunk 16 above)
Comment (ryliemn) [reply jQuery/styles.css]: I think we might not be on the same page here. I'm not trying to center anything at the moment, my menu is `position: fixed` to the left side of the window at a static width of 240px, so the `margin-left: 240px` here is just to ensure nothing in `#content` overlaps with the menu. The intent of the padding here wasn't to center the content, just to give the `Hello wolr` some breathing room and will almost certainly change once we start working on the actual content of the page

File: jQuery/honeyHeadquarters.js (hunk 13 above)
Comment (stffrd) [reply jQuery/honeyHeadquarters.js]: Move the entire block, because we're eventually going to put the functions in their own file as well.

File: jQuery/styles.css (hunk 16 above)
Comment (stffrd) [reply jQuery/styles.css]: Ah. I'm not exactly running this locally just yet, so it's hard for me to know what it's supposed to look like. my b. Still something to think about though

**Comment by ryliemn** [comment]
fixed more stuff. I think we can probably go ahead and merge this in and continue on a new PR cause this one is getting pretty cluttered haha
